tools/block-generator/generator/generate.go

```diff
@@ -359,21 +377,27 @@ func (g *generator) finishRound(txnCount uint64) {
 
 // WriteBlock generates a block full of new transactions and writes it to the writer.
 func (g *generator) WriteBlock(output io.Writer, round uint64) error {
-
-	if round != g.round {
-		fmt.Printf("Generator only supports sequential block access. Expected %d but received request for %d.\n", g.round, round)
+	if round < g.dbround {
```

Comment:
we actually need to pass round into WriteBlock() for generating blocks for a preloaded db. generator.round would be different from round in this case. A preloaded db may be at round 10, and the request for next round would be /v2/blocks/11. But the generator has to start at round 0 and we add each block to the in-mem ledger.

Reply:
The more and more I look at the block generator code, the more I keep asking myself why the block generator needs to be stateful or keep track of genesis and previous block hash. Does indexer need to maintain blockchain consistency for the purposes of performance testing? Isn't it going to simply trust the block that it gets from algod and save it no questions asked? Or maybe it's actually going to compute a block's hash and refuse to save the next block if its previous block hash doesn't match. If the latter, then the current statefulness is definitely required. But if it's taking the block from algod -no questions asked (except for say round number), then we really don't need all the extra data, and `WriteBlock()` need not take `round` as input while `getBlockHandler`'s anonymous function can simple discard or ignore the round that it gets from the http request.

Reply:
Reading more I see that `WriteBlock()` calls `g.ledger.AddBlock()` and that this indeed cares that the expected round number be provided. But my hunch is that we should still be able to totally ignore the round provided in the http request, and just ensure the the block generator remains consistent with the ledger- a problem that is already solved.

Reply:
We definitely need to make sure the block is updated to the requested round before returning it, otherwise Indexer would fail on primary key constraints when trying to insert a block that already exists. Other than that, I don't think the round is especially important.